lib/trianglify.js

```diff
@@ -108,11 +107,11 @@ function Trianglify(opts) {
 
   // Set up normalizers
   var norm_x = function(x) {
-    return _map(x, [-bleed_x, width+bleed_x], [0, 1]);
+    return _map(x, [0, width], [0, 1]);
```

Comment:
The normalization here previously returned a float strictly within the interval `[0, 1]` (inclusive), and it's possible that API consumers depend on this behavior in their `color_function` implementation.

With the changes here, we'll be passing numbers outside that range to the `color_function`: 

<img width="636" alt="screen shot 2018-10-18 at 9 17 20 am" src="https://user-images.githubusercontent.com/347189/47168960-223be980-d2b7-11e8-98c4-f0e343af3d3d.png">

While `chroma.interpolate` is able to handle this edge case, a 3rd party `color_function` may not be.

I'm on-board with changing the normalization function to start and end the gradient in the viewport, but I'd like to clamp out-of-bounds values to 0 or 1. Here's a rough sketch of how that might look:

```
var norm_x = function(x) {
  return _clamp(
    _map(x, [0, width], [0, 1]),
    [0, 1]
  )
}

// and here's my suggested implementation for clamp() (should go right above the function definition for _map)
function _clamp (num, interval) {
  return Math.min(
    Math.max(num, interval[0]),
    interval[1]
  )
}
```

Reply:
Ooh bummer! You are totally right, I overlooked the details of _map, I assumed it was already the case.